webpack/assets/javascripts/react_app/routes/PluginRoutes.js

```diff
@@ -0,0 +1,26 @@
+import { Route } from 'react-router-dom';
+import React from 'react';
+import { addGlobalFill } from '../components/common/Fill/GlobalFill';
+import { renderRoute } from './RoutingService';
+
+/**
+ * Adds a plugin's routes into core
+ * @param  {String} id  plugin's id - can be its name
+ * @param  {Array}   routes an array that contains a plugin's routes
+ */
+export const registerRoutes = (id, routes) =>
+  routes.map(
+    ({ render, path, beforeRendering = undefined, ...routeProps }, index) =>
+      addGlobalFill(
```

Comment:
can the plugin manipulate the weight?
I'd imagine we could have core routes with weight `> 10000` and plugins could register by default with lower weight, but would be allowed to override the core route by defining the weight higher for the specific route.

Reply:
`addGlobalFill` doesn't get any weight argument, so the registered routes cannot change their location.
Route component also has an `exact` prop which makes the routes order redundant (the switcher will pick the exact route regardless the ordering), IMHO allowing routes ordering can give us a lot of issues between plugins (also an area that we don't have good coverage), using `exact` routes should avoid this from happening.

Reply:
make sense :+1: once we will have some duplicate entries, we can think about it.
Although duplicate entries could be useful for plugins in some cases :shrug: